src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocket.java

```diff
@@ -29,21 +30,39 @@ public RxWebSocket(OkHttpClient http) {
             em -> {
               Request rq = new Request.Builder().url(url).build();
 
-              ws = http.newWebSocket(rq, new Listener(em));
+              ws = http.newWebSocket(rq, new Listener(RxWebSocket.this, em));
             })
+        .takeUntil(e -> e instanceof RxWebSocketEvent.Closed)
         .doOnNext(e -> LOG.trace("Received: {}.", e))
         .doOnError(e -> LOG.warn("Error: {}.", e));
   }
 
+  public void close(int code) {
+    close(code, null);
+  }
+
+  public void close(int code, String reason) {
+    ws.close(code, reason);
+  }
+
+  public Completable closeDueToInvalidSession() {
```

Comment:
This is a leaky abstraction. Invalid session is a discord concept. The RxWebSocket class shouldn't care whether it's being used in discord or not.

Reply:
yeah you're right, I'll remove it and push it into gateway